### PR TITLE
[pytorch] [build] PT1.11.0 - reduce PyTorch GPU Train image size 

### PR DESCRIPTION
**Pull request by ydaiming**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Description

Reduce PyTorch GPU Train image size by moving `torchnet` installation to each `AWS-PyTorch` installation layer, avoid redundant official PyTorch installation stored in the previous `pip install` layer.

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [ ] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pytorch/training/docker/1.11/py3/cu115/Dockerfile.gpu [hunk 1]
```diff
@@ -232,15 +230,9 @@ ARG PT_TORCHAUDIO_URL
...
-# Re-install AWS-PyTorch binaries
-RUN pip uninstall -y torch \
- && pip install --no-cache-dir -U ${PT_E3_TRAINING_URL} \
- && pip uninstall -y torchvision \
- && pip install --no-deps --no-cache-dir -U ${PT_TORCHVISION_URL} \
- && pip uninstall -y torchaudio \
- && pip install --no-deps --no-cache-dir -U ${PT_TORCHAUDIO_URL} \
- && pip uninstall -y torchdata \
- && pip install --no-deps --no-cache-dir -U ${PT_TORCHDATA_URL}
+# Install AWS-PyTorch and other torch packages
```
Comment (tejaschumbalkar) [inline pytorch/training/docker/1.11/py3/cu115/Dockerfile.gpu:233]: Removing and installing packages in such a way contributes to decrease in size?

**Review by tejaschumbalkar — COMMENTED** [review]
Can we also make sure that the `apt-get clean` command is executed at every apt installation step, especially the last step like https://github.com/aws/deep-learning-containers/pull/1783/files#diff-c9c2cb65fddee89f68849e7ec1adf42b4b1af5830912ccfe1ef908a1a2e90867R402

This will make sure that the apt cache is cleared and contribute to decrease in image size even further.

**Comment by tejaschumbalkar** [comment]
Can we also update the PR description

File: pytorch/training/docker/1.11/py3/cu115/Dockerfile.gpu (hunk 1 above)
Comment (ydaiming) [reply pytorch/training/docker/1.11/py3/cu115/Dockerfile.gpu:233]: the main source of decrease comes from moving `torchnet` to this step.
Collapsing the commands makes it better readability.

**Comment by tejaschumbalkar** [comment]
Do we need to change the image baseline size in buildspec.yml?

**Comment by tejaschumbalkar** [comment]
> image



> Do we need to change the image baseline size in buildspec.yml?

This needs to be addressed in the dedicated cu115 buildspec file created as a part of #1795

**Comment by tejaschumbalkar** [comment]
> image



> Do we need to change the image baseline size in buildspec.yml?

This needs to be addressed in the dedicated cu115 buildspec file created as a part of #1795

**Comment by tejaschumbalkar** [comment]
> Can we also update the PR description



> Can we also update the PR description

PR LGTM. Lets update the PR description and prepare for merge.

**Comment by tejaschumbalkar** [comment]
> Can we also update the PR description



> Can we also update the PR description

PR LGTM. Lets update the PR description and prepare for merge.

**Review by tejaschumbalkar — APPROVED** [review]

**Pull request merged**